### PR TITLE
animates checkmark on save button on successful save.

### PR DESCRIPTION
**Pull request by tdowds**
So the goal of this commit is to add a checkmark animation that occurs on a button whenever pressing that button results in a successful operation. 

This is related to #495 

[source code for animation](http://codepen.io/drewbkoch/pen/ogyXEK) for reference.
[source code for checkmark](http://www.cssflow.com/snippets/check-buttons/demo/scss)

### CODE REVIEW

File: huxley/www/js/components/Button.js [hunk 1]
```diff
@@ -40,11 +42,12 @@ var Button = React.createClass({
...
           'button-red': this.props.color == 'red',
           'rounded-small': true,
-          'loading': this.props.loading
+          'loading': this.props.loading,
+          'button-checkmark': this.props.success
         })}
         disabled={this.props.loading}
         to={this.props.href}>
-        <span>{this.props.children}</span>
+          <span>{this.props.children}</span>
```
Comment (kmeht) [inline huxley/www/js/components/Button.js]: revert

File: huxley/www/js/components/Button.js [hunk 2]
```diff
@@ -40,11 +42,12 @@ var Button = React.createClass({
           'button-yellow': this.props.color == 'yellow',
           'button-red': this.props.color == 'red',
           'rounded-small': true,
-          'loading': this.props.loading
+          'loading': this.props.loading,
+          'button-checkmark': this.props.success
```
Comment (kmeht) [inline huxley/www/js/components/Button.js]: nit: trailing comma

File: huxley/www/js/components/Button.js [hunk 3]
```diff
@@ -15,13 +15,15 @@ var Button = React.createClass({
     href: React.PropTypes.string,
     loading: React.PropTypes.bool,
     size: React.PropTypes.oneOf(['small', 'medium', 'large']),
+    success: React.PropTypes.bool
```
Comment (kmeht) [inline huxley/www/js/components/Button.js]: nit: trailing comma,

File: huxley/www/js/components/AdvisorProfileView.js [hunk 4]
```diff
@@ -364,8 +368,13 @@ var AdvisorProfileView = React.createClass({
   _handleSuccess: function(data, status, jqXHR) {
     this.setState({
       errors: {},
-      loading: false
+      loading: false,
+      success: true
     });
+
+    setTimeout(() => {
```
Comment (kmeht) [inline huxley/www/js/components/AdvisorProfileView.js]: We should store this timeout and cancel it if the component unmounts, and also if the user clicks submit again.

File: huxley/www/js/components/AdvisorProfileView.js [hunk 5]
```diff
@@ -342,6 +345,7 @@ var AdvisorProfileView = React.createClass({
   },
 
   _handleSubmit: function(event) {
+    this.setState({loading: true});
```
Comment (kmeht) [inline huxley/www/js/components/AdvisorProfileView.js]: Whoops, didn't know we forgot to do this :)

File: huxley/www/js/components/AdvisorProfileView.js [hunk 6]
```diff
@@ -51,7 +51,9 @@ var AdvisorProfileView = React.createClass({
       primary_phone: school.primary_phone,
       secondary_name: school.secondary_name,
       secondary_email: school.secondary_email,
-      secondary_phone: school.secondary_phone
+      secondary_phone: school.secondary_phone,
+      loading: false,
+      success: false
```
Comment (kmeht) [inline huxley/www/js/components/AdvisorProfileView.js]: nit: trailing comma

**Comment by tdowds** [comment]
If there aren't any other substantive changes left, I'm going to replicate this behavior in other parts of the app where it makes sense and then try to get the animation working.

**Comment by kmeht** [comment]
The timeout ID doesn't have to be component state, it can just be an instance property on the component, very similar to how we handle Store subscriptions, e.g.
```js
this._successTimout && clearTimeout(this._successTimeout);
this._successTimeout = setTimeout(() ==> this.setState({success: false}));
```

**Pull request merged**